processor/src/main/java/org/mapstruct/ap/internal/model/BeanMappingMethod.java

```diff
@@ -255,17 +255,19 @@ private boolean handleDefinedSourceMappings() {
                         resultPropertyName = first( targetRef.getPropertyEntries() ).getName();
                     }
 
-                    if (!unprocessedTargetProperties.containsKey( resultPropertyName ) ) {
+                    if ( !unprocessedTargetProperties.containsKey( resultPropertyName ) ) {
                          boolean hasReadAccessor =
                             method.getResultType().getPropertyReadAccessors().containsKey( mapping.getTargetName() );
-                        ctx.getMessager().printMessage( method.getExecutable(),
-                            mapping.getMirror(),
-                            mapping.getSourceAnnotationValue(),
-                            hasReadAccessor ? Message.BEANMAPPING_PROPERTY_HAS_NO_WRITE_ACCESSOR_IN_RESULTTYPE :
-                                Message.BEANMAPPING_UNKNOWN_PROPERTY_IN_RESULTTYPE,
-                            mapping.getTargetName()
-                        );
-                        errorOccurred = true;
+                        if ( !hasReadAccessor || !mapping.isIgnored() ) {
```

Comment:
Can we de-compose this into two separate ifs? Together with the second one on `hasReadAccessor` inline I find i a bit tough to digest.